### ISSUE TITLE
Integrate login with Instagram

### ISSUE THREAD

**Issue by VenkatVenkatesh** [DashboardHub/TribeDashboard#23]
#2

**Comment by VenkatVenkatesh** [comment]
Hi @eddiejaoude,
Based on the following reference, For Instagram login, https://firebase.googleblog.com/2016/10/authenticate-your-firebase-users-with.html.
I think we need to create a cloud function to get the `oAuth credentials` of the instagram account using which we can do a customLogin in firebase. 
Please let me know your thoughts on this approach ?

**Comment by eddiejaoude** [comment]
Hi @VenkatVenkatesh , that is quite old and I think out of date. After a quick read of the official docs I found the following...

> To continuously improve Instagram users' privacy and security, we are accelerating the deprecation of Instagram API Platform

According to the Instagram [docs](https://www.instagram.com/developer/), their specific API has been deprecated and I think no longer works as of December 2018. 

> This document explains how to successfully call the Instagram Graph API

It seems that their new version of their API has been brought into Facebook API, here are their [docs](https://developers.facebook.com/docs/instagram-api/getting-started/), which should make things easier.

@SrijithRad what do you think?

**Comment by SrijithRad** [comment]
Agreed! Hopefully it is straight forward integrating with firebase. From the looks of it, it does seem so.

**Comment by VenkatVenkatesh** [comment]
@eddiejaoude Based on the following [graph api](  https://developers.facebook.com/docs/instagram-api/getting-started ) we can implement instagram login for only for business accounts using graph API.

I connected facebook account with instagram and tried to use their graph API. But the data field was empty. Could you please let me know your thoughts on this ?

**Comment by eddiejaoude** [comment]
If it only works for business accounts, then that is fine, we will have to let the user know with some text that it is a business instagram account only

**Comment by VenkatVenkatesh** [comment]
Hi @eddiejaoude Here are the findings for instagram login.

- The facebook graph API can only be used for  instagram business accounts.
- The instagram business account should be connected to a facebook page. 
`FYI: https://developers.facebook.com/docs/instagram-api/overview#pages`
- The instagram business account should be connected to facebook business account. 
`FYI: https://stackoverflow.com/questions/38929890/how-to-use-facebook-graph-api-to-read-instagram-user`
- To get information about ordinary instagram accounts we cannot use facebook API. We have to use Instagram API

**Comment by eddiejaoude** [comment]
> To get information about ordinary instagram accounts we cannot use facebook API. We have to use Instagram API

Ok thank you. I am not sure I understand the last point, I thought it wasn't possible for ordinary instagram accounts?

**Comment by VenkatVenkatesh** [comment]
you are correct @eddiejaoude . It is not possible to use facebook graph API for ordinary instagram accounts. We can only use the legacy Instagram API.

**Comment by eddiejaoude** [comment]
Ok, the legacy API they will eventually remove I guess? So best just to do it for the business Instagram accounts

**Comment by SrijithRad** [comment]
> Hi @VenkatVenkatesh , that is quite old and I think out of date. After a quick read of the official docs I found the following...
> 
> > To continuously improve Instagram users' privacy and security, we are accelerating the deprecation of Instagram API Platform
> 
> According to the Instagram [docs](https://www.instagram.com/developer/), their specific API has been deprecated and I think no longer works as of December 2018.
> 
> > This document explains how to successfully call the Instagram Graph API
> 
> It seems that their new version of their API has been brought into Facebook API, here are their [docs](https://developers.facebook.com/docs/instagram-api/getting-started/), which should make things easier.
> 
> @SrijithRad what do you think?

Agreed @eddiejaoude . Makes sense. No use integrating with a deprecated api.
